gapic/src/main/com/google/gapid/widgets/LinkifiedTree.java

```diff
@@ -411,9 +420,25 @@ private boolean shouldIgnoreColors(Event event) {
       return (event.detail & SWT.SELECTED) != 0;
     }
 
+    public void updateHierarchy(TreeItem item) {
```

Comment:
This is way to expensive. You update the item and its children. Then you recurse with the parent item, which will again, update all the items that have already been updated. And this will happen again for every ancestor...
In fact, calling this function with any item will end up updating the entire tree, so why pass in an item at all?

Reply:
The problem is we actually need to update the entire visible tree especially in cases of disabling/enabling a command with children. In our luck without secondary command buffer, this is a tree with the depth of 2 and with secondary command buffer it's with a depth of 4. 

Still, this is an expensive operation but the actual operation triggers a new replay that will take at least seconds so I don't think that this is a bottleneck here. In terms of UI, in my laptop I couldn't see any issues.

To be able to improve this, when we started a disable experiment, we need to calculate and return which commands are actually affected and only update their children & parents. Which is doable but I am not sure performance concern justifies the complexity. 

I would suggest waiting if anyone actually is affected by UI unresponsiveness from this and if they do then we can optimize but currently it's good enough.

Reply:
My point is that you update most items many times. Each time you recurse into a parent, you re-update the entire part of the tree you just updated. So, if you need to update the entire tree, fine, but only update it once.